### ISSUE TITLE
Internal Server error with German Language on the Website

### ISSUE THREAD

**Issue by Voltoviper** [odoo/odoo#10607]
I guys, 

i got an Internal Server Error, when i turn on the german Language in the Website Admin. Even if the Main Language is german... can you help me? 

Here the Stacktrace

```
500: Internal Server Error

Fehler

Error message:

xmlParseEntityRef: no name, line 6, column 42
Zurückverfolgen

Traceback (most recent call last):
  File "/opt/odoo/odoo/addons/website/models/ir_http.py", line 242, in _handle_exception
    response = super(ir_http, self)._handle_exception(exception)
  File "/opt/odoo/odoo/openerp/addons/base/ir/ir_http.py", line 157, in _handle_exception
    return request._handle_exception(exception)
  File "/opt/odoo/odoo/openerp/http.py", line 781, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo/openerp/addons/base/ir/ir_http.py", line 182, in _dispatch
    result = request.dispatch()
  File "/opt/odoo/odoo/openerp/http.py", line 840, in dispatch
    r = self._call_function(**self.params)
  File "/opt/odoo/odoo/openerp/http.py", line 316, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/odoo/openerp/service/model.py", line 118, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/odoo/openerp/http.py", line 309, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/odoo/openerp/http.py", line 959, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/odoo/openerp/http.py", line 509, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/odoo/addons/website/controllers/main.py", line 44, in index
    return request.registry['ir.http'].reroute(first_menu.url)
  File "/opt/odoo/odoo/addons/website/models/ir_http.py", line 209, in reroute
    return self._dispatch()
  File "/opt/odoo/odoo/addons/utm/models/ir_http.py", line 13, in _dispatch
    response = super(ir_http, self)._dispatch()
  File "/opt/odoo/odoo/addons/website/models/ir_http.py", line 190, in _dispatch
    resp = super(ir_http, self)._dispatch()
  File "/opt/odoo/odoo/addons/web_editor/models/ir_http.py", line 16, in _dispatch
    return super(ir_http, self)._dispatch()
  File "/opt/odoo/odoo/openerp/addons/base/ir/ir_http.py", line 186, in _dispatch
    return self._handle_exception(e)
  File "/opt/odoo/odoo/addons/website/models/ir_http.py", line 279, in _handle_exception
    context=request.context)
  File "/opt/odoo/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/opt/odoo/odoo/addons/website/models/ir_ui_view.py", line 85, in _views_get
    result += self._views_get(cr, uid, called_view, options=options, bundles=bundles, context=context)
  File "/opt/odoo/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/opt/odoo/odoo/addons/website/models/ir_ui_view.py", line 98, in _views_get
    context=context, root=False):
  File "/opt/odoo/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/opt/odoo/odoo/addons/website/models/ir_ui_view.py", line 98, in _views_get
    context=context, root=False):
  File "/opt/odoo/odoo/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/opt/odoo/odoo/addons/website/models/ir_ui_view.py", line 75, in _views_get
    node = etree.fromstring(view.arch)
  File "lxml.etree.pyx", line 3032, in lxml.etree.fromstring (src/lxml/lxml.etree.c:68106)
  File "parser.pxi", line 1785, in lxml.etree._parseMemoryDocument (src/lxml/lxml.etree.c:102455)
  File "parser.pxi", line 1666, in lxml.etree._parseDoc (src/lxml/lxml.etree.c:101214)
  File "parser.pxi", line 1035, in lxml.etree._BaseParser._parseUnicodeDoc (src/lxml/lxml.etree.c:96124)
  File "parser.pxi", line 582, in lxml.etree._ParserContext._handleParseResultDoc (src/lxml/lxml.etree.c:91275)
  File "parser.pxi", line 683, in lxml.etree._handleParseResult (src/lxml/lxml.etree.c:92461)
  File "parser.pxi", line 622, in lxml.etree._raiseParseError (src/lxml/lxml.etree.c:91757)
XMLSyntaxError: xmlParseEntityRef: no name, line 6, column 42
```

**Comment by leoniscsem** [comment]
Can confirm, also affected by this error.

**Comment by lukebranch** [comment]
@leoniscsem @Voltoviper ,

I believe this needs to be fixed in transifex. There was a problem with the XML syntax with some of the translations which will cause these XML Syntax errors that you see here.

You can see a similar problem reported here

**Comment by leoniscsem** [comment]
Great tool, but in this case not necessary since the step is just to activate the Website Builder module and enabling the German language pack - immediate 500.

**Comment by lukebranch** [comment]
@leoniscsem ,

I'd recommend finding one of the most active threads about this topic (malformed XML in translation files) and report there for better visibility.

This seems to be a common issue with the version 9.0 translations. I believe Odoo SA are working on it, however to expedite the process with the German language pack i'd suggest finding one of those more active threads and joining the discussion there.

@mart-e , this appears to be another issue that needs to be fixed in transifex due to improperly formed XML.

**Comment by mart-e** [comment]
Hello,

Indeed I have been working on fixing wrongly translated planners (see [this message on the forum](https://www.odoo.com/forum/help-1/question/greek-translation-bug-in-sales-menu-95169#answer_95229) for explanations and ways to fix it).

The proper way is to fix it on Transifex.

However, yesterday, I made another fix, rev 6872aae8ef9c1a1fc6a8283aa9d12ce3a6859157, that actually hides the error in case of a wrongly translated views.
This way, we don't blame the users for the errors of translators ;-)

**Comment by lukebranch** [comment]
@mart-e ,

Thanks for the quick reply!

**Comment by leoniscsem** [comment]
This did the trick, website does not deliver 500 anymore. Awesome, thanks!

**Comment by mart-e** [comment]
But then, could be good to check which translation is actually wrong and try to fix it

**Comment by leoniscsem** [comment]
There are still some funny terms in the German translation, so I will definitely go and check out Transifex. (Yet not within a couple of days, we are currently busy with fixing some other quirks in our testing environment.)

**Comment by mart-e** [comment]
@leoniscsem I made a test to try to detect all errors in translations views and log them.
You can see the results [on runbot](http://runbot.odoo.com/runbot/repo/git-github-com-odoo-dev-odoo-2?search=9.0-plannertestlang-mat) (I will try to rebase it weekly to check with latest translations) or see the [result of last build](http://runbot.odoo.com/runbot/build/127374) (taking ~30 hours to test)

**Comment by wtaferner** [comment]
@mart-e 
Thank you for providing the test...I have fixed all issues which I could identify easily from the full log from a former test run for the language de_DE.

**Comment by mart-e** [comment]
@wtaferner I my previous test, I only checked the planner translations (as it was breaking when clicking on menus). In this latest, I test all QWeb views (which makes about 200 remaining errors, at least 28 in German translations so far).

**Comment by wtaferner** [comment]
Jep, I checked the result of the last build and I guess I got them all...we will see on the next build/testrun ;-)

**Comment by wtaferner** [comment]
@mart-e 
Would you eventually provide some kind of equal log in case your automatic escape mechanism is triggered? Based on this the wrong translation is always logged, in case triggered even though the user is not blocked.

**Comment by mart-e** [comment]
@wtaferner I guess we can log them in debug or warning but I first want to make sure the escaping works properly.

**Comment by wtaferner** [comment]
Ok, sounds good, let's see if it works out...

**Comment by wtaferner** [comment]
@mart-e 
It seems your escape mechanism does somehow escape too many translated terms for some use cases (Tabs of form view).

http://129807-9-0-f2146d.runbot9.odoo.com/web/?debug#id=14&view_type=form&model=res.partner&menu_id=154&action=169

**Comment by mart-e** [comment]
Hmm should have been fixed at d1abc952ccdc74c3e7f47f4c30ed1b5ca3149633, will check it thanks
